verify/verify.go

```diff
@@ -866,20 +866,52 @@ func verifyHash256(quote *pb.QuoteV4) error {
 	return nil
 }
 
-func isSvnHigherOrEqual(svn []byte, components []pcs.TcbComponent) bool {
-	for i := range svn {
-		if svn[i] < components[i].Svn {
+func isCPUSvnHigherOrEqual(pckCertCPUSvnComponents []byte, sgxTcbcomponents []pcs.TcbComponent) bool {
+	for i := range pckCertCPUSvnComponents {
+		if pckCertCPUSvnComponents[i] < sgxTcbcomponents[i].Svn {
 			return false
 		}
 	}
 	return true
 }
 
+func isTdxTcbSvnHigherOrEqual(teeTcbSvn []byte, tdxTcbcomponents []pcs.TcbComponent) bool {
+	start := 0
+	if teeTcbSvn[0] > 0 {
```

Comment:
As per documentation, if TeeTcbSvn[1] value is >= 1 then comparison should be done starting index 2 instead of 0.

Reply:
As per your comment you should be using teeTcbSvn[1] but in the code you have used teeTcbSvn[0] for the check.

Reply:
Good catch!

Reply:
Can you add a testcase for this?
We have some sample API responses in testing/testdata which are failing collateral verification due to OutofDate status, you can add new API response and your changes should give UpToDate status.

Reply:
But you'll need to add a new TDX quote.  Will a quote generated on one of our machines be suitable? @ivishwesh

Reply:
> Can you add a testcase for this? We have some sample API responses in testing/testdata which are failing collateral verification due to OutofDate status, you can add new API response and your changes should give UpToDate status.

I didn't see any failure in tests. How did you confirm the failure, can you elaborate on it?

Reply:
While writing tests earlier we were having a quote which was OutofDate acc. to TcbInfo API response so I added a testcase which expects failure. Now you can add a new quote and a new sample API response(in testdata) and your changes should return status as UpToDate.
(@ivishwesh can confirm whether we can add a new TDX quote in testdata generated from VIT mach)
https://github.com/google/go-tdx-guest/blob/686ff34667713c7b1a2a1d23b6c323d4db5611cb/verify/verify_test.go#L676

Reply:
We discussed we will add new quote and unit tests in a following PR.